### ISSUE TITLE
openlayers 7.2.2.  Load Baidu map tiles in the display of the upper part of a zoom under

### ISSUE THREAD

**Issue by zhenglei01** [openlayers/openlayers#14507]
projection in view will be made in 84 frame. It will be the custom bd09 frame used by Baidu map data source. Tiles will be different when zoom equals 5

Added Baidu coordinate system, tileUrlFunction z has been changed to -tileCoord[2]-1;
 In view, if projection is set to bd09 or 3857, the different zoom maps will display correctly, but when switched to 4326 and zoom equals 5,, the top and bottom of the map will load tiles at two [scales.]

![1676357079549](https://user-images.githubusercontent.com/88827883/218659927-c70a42e0-9742-4016-9710-e71d7c2820da.jpg)

![image](https://user-images.githubusercontent.com/88827883/218660340-5b006973-4a51-42b6-a67f-77aa0513af7e.png)

![image](https://user-images.githubusercontent.com/88827883/218660190-8f7ddecc-efb2-4799-9ff5-65c37e7b0fc8.png)

![image](https://user-images.githubusercontent.com/88827883/218660528-80b425af-2069-4cb4-813e-f9a5cdffb878.png)

This is the picture above. hope the boss will help to find out what the reason is and how to deal with it.

**Comment by zhenglei01** [comment]
I just tried another version, and I have the same problem. I don't know if it's because of my configuration.

**Comment by mike-000** [comment]
It also happens when reprojecting OSM to EPSG:4326

![image](https://user-images.githubusercontent.com/49240900/220200513-e3ea0f53-6c01-4000-acdb-cf0035d0ea23.png)

It is a consequence of the two projections having different resolutions as you move away from the equator, so tiles with the resolution most suitable for the latitude are used.

**Comment by zhenglei01** [comment]
> It also happens when reprojecting OSM to EPSG:4326
> 
> ![image](https://user-images.githubusercontent.com/49240900/220200513-e3ea0f53-6c01-4000-acdb-cf0035d0ea23.png)
> 
> It is a consequence of the two projections having different resolutions as you move away from the equator, so tiles with the resolution most suitable for the latitude are used.


  Thanks for the answer, but is this normal? It's not like this in any other coordinate system, it's not a good visual comparison

**Comment by mike-000** [comment]
Yes, it is normal.  An alternative would be to limit the tile zoom to the view zoom at each level such as in https://codesandbox.io/s/reprojection-wgs84-forked-8h554o?file=/main.js where there is a continuous gradual reduction in text size as you move away from the equator instead of the steps where a new tile resolution is used to compensate for the difference in projection resolution.

**Comment by zhenglei01** [comment]
> 

this is creating super many layers, and when you use more layers, will it affect the performance in the later use?

**Comment by zhenglei01** [comment]
> Yes, it is normal. An alternative would be to limit the tile zoom to the view zoom at each level such as in https://codesandbox.io/s/reprojection-wgs84-forked-8h554o?file=/main.js where there is a continuous gradual reduction in text size as you move away from the equator instead of the steps where a new tile resolution is used to compensate for the difference in projection resolution.

this is creating super many layers, and when you use more layers, will it affect the performance in the later use?

**Comment by ahocevar** [comment]
You could also use `useGeographic()`, then geographic coordinates will be used for the resolution matching, and this effect will go away. See https://codesandbox.io/s/reprojection-wgs84-forked-t5qye8?file=/main.js.
